### DISCUSSION TITLE
Russian roulette of e- after brems event with bcse

### DISCUSSION THREAD

**Discussion by mainegra** [clrp-code/egs_brachy#56] [General]
I was pleasantly surprised to discover that `egs_brachy` has the ability to do UBS and BCSE!
Looking at the method `EB_Application::doPhotonSplitting` in egs_brachy.cpp I noticed that after a brems event, if bcse is on and the electron loses the Russian roulette game, rather than killing the electron, it is turned into a _"regular"_ electron that can be traced normally.

Am I interpreting this correctly?

**Comment by mchamberland** [comment]
It's been years since I've looked at that part of the code, but I think you're misinterpreting it.

I assume you're referring to [this section of the code.](https://github.com/clrp-code/egs_brachy/blob/0219903717c0c759a7a60c549cb8850fb7455334/egs_brachy/egs_brachy.cpp#L1877-L1887)  Here, the reference to Russian Roulette in the comment is referring to how the energy of the electron is only _randomly_ decreased in BCSE (I had to look up [Ali and Dave's BCSE paper](https://doi.org/10.1118/1.2736778) to refresh my memory on this!). The actual Russian Roulette that kills electrons still happens normally within the EGSnrc macros.

**Reply by mainegra** [reply]
My understanding of how egs_brachy does BCSE + UBS is this:

- UBS is leveraged from EGSnrc: 
  -   There is no actual RR-ing of electrons in the BREMS routine. 
  -   One electron is kept with its energy reduced by the energy of the last brems photon. See routine BREMS in egsnrc.mortran.
  -   This is implemented so for efficiency purposes
  -   It is an approximation, correct only on average after a large number of histories
- egs_brachy deals with everything related to BCSE:
  - routine egs_bcse in egs_brachy.macros does the actual enhancement of the brems xsections
  - method EB_Application::doPhotonSplitting takes care of BCSE by:
    -     playing a RR game with the fat electron from UBS with a 1/fenh probability
    -     if e- loses RR game its energy is increased by the energy of the photon on top of the stack
- egs_brachy also takes care of splitting fluorescent photons after atomic relaxations when UBS is used in EB_Application::doPhotonSplitting. This was not implemented in EGSnrc (as it should!) and hence must be taken care of by applications using EGSnrc-intrinsic UBS. egs_brachy does it well by checking that the photon is actually fat in case EGSnrc implements the splitting in the future. ;-)

If this is correct, I was wondering what it means to restore the e- energy. Usually after a RR game is lost, the particle is killed. Here it seems it is kept with its initial energy.

_I am not saying this is wrong, I am just trying to understand it!_

**Reply by mchamberland** [reply]
Yes, I see your confusion.

If I understand correctly myself, I believe the _comment_ is misleading: it is not a Russian Roulette game being played here; it's just part of the BCSE algorithm that randomly decreases the electron energy.

But within regular EGSnrc, after a brems event, the electron energy is always decreased by the photon energy. So, within `egs_brachy`, we _add_ the energy back to the electron with probability (1 - 1/bcse_factor). That's equivalent to decreasing the electron energy with probability 1/bcse_factor, as per the BCSE paper.

**Reply by mainegra** [reply]
Ah yes! Thanks for clarifying this Mark! Sorry for dragging you down memory lane!
Would you agree with the points I made above in the bullet points? Meaning:
- VRTs for modelling x-ray targets in egs_brachy are not simply left to EGSnrc, but the application needs to take care of certain subtleties of the algorithms as they are implemented in EGSnrc?
- BCSE is actually taken care of in egs_brachy except for the RR-ing of higher order particles after photon interactions, which uses EGSnrc internal RR.

BTW, looking here at the UBS implementation made me realize that the radiative splitting AO needs to take care of splitting fluorescent photons after atomic relaxations! #urgent-todo

**Reply by mchamberland** [reply]
Yes, I believe you are correct with those bullet points.
